src/isaac.rs

```diff
@@ -516,6 +522,142 @@ impl fmt::Debug for Isaac64Rng {
     }
 }
 
+#[cfg(feature="serde-1")]
+mod rand_size_usize_serde {
+    use super::RAND_SIZE_USIZE;
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+    use serde::de::{Visitor,SeqAccess};
+    use serde::de;
+
+    use std::num::Wrapping;
+    use std::fmt;
+
+    pub fn serialize<T, S>(arr: &[Wrapping<T>;RAND_SIZE_USIZE], ser: S) -> Result<S::Ok, S::Error> 
+    where
+        T: Serialize,
+        S: Serializer 
+    {
+        use serde::ser::SerializeTuple;
+
+        let mut seq = ser.serialize_tuple(RAND_SIZE_USIZE)?;
+
+        for e in arr.iter() {
+            seq.serialize_element(&e)?;
+        }
+
+        seq.end()
+    }
+
+    #[inline]
+    pub fn deserialize<'de, T, D>(de: D) -> Result<[Wrapping<T>;RAND_SIZE_USIZE], D::Error>
+    where
+        T: Deserialize<'de>+Default+Copy,
+        D: Deserializer<'de>,
+    {
+        use std::marker::PhantomData;
+        struct ArrayVisitor<T> {
+            _pd: PhantomData<T>,
+        };
+        impl<'de,T> Visitor<'de> for ArrayVisitor<T>
+        where
+            T: Deserialize<'de>+Default+Copy
+        {
+            type Value = [Wrapping<T>; RAND_SIZE_USIZE];
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("Isaac state array")
+            }
+
+            #[inline]
+            fn visit_seq<A>(self, mut seq: A) -> Result<[Wrapping<T>; RAND_SIZE_USIZE], A::Error>
+            where
+                A: SeqAccess<'de>,
+            {
+                let mut out = [Wrapping(Default::default());RAND_SIZE_USIZE];
+
+                for i in 0..RAND_SIZE_USIZE {
+                    match seq.next_element()? {
+                        Some(val) => out[i] = val,
+                        None => return Err(de::Error::invalid_length(i, &self)),
+                    };
+                }
+
+                Ok(out)
+            }
+        }
+
+        de.deserialize_tuple(RAND_SIZE_USIZE, ArrayVisitor{_pd: PhantomData})
+    }
+}
+
+#[cfg(feature="serde-1")]
+mod rand_size_64_serde {
+    use super::RAND_SIZE_64;
```

Comment:
Isn't this module identical to the `usize` version? I doubt we'll be changing `RAND_SIZE_USIZE` or `RAND_SIZE_64` — they're fundamental to the algorithms — so it seems silly duplicating the code here.

Reply:
Yeah, I thought the same thing, but I did it just to be safe. I can remove it.

Reply:
`RAND_SIZE_USIZE` was just there to avoid a few casts. In https://github.com/dhardy/rand/pull/4/files I remoced it.